fl4health/clients/basic_client.py

```diff
@@ -612,13 +630,16 @@ def train_by_epochs(
                 Loss is a dictionary of one or more losses that represent the different components of the loss.
         """
         self.model.train()
-        local_step = 0
+        steps_this_round = 0  # Reset number of steps this round
         for local_epoch in range(epochs):
             self.train_metric_manager.clear()
             self.train_loss_meter.clear()
             # update before epoch hook
             self.update_before_epoch(epoch=local_epoch)
-            for input, target in self.train_loader:
+            # Print initial log string on epoch start, not epoch end
+            self._handle_logging({}, {}, current_round, local_epoch)
```

Comment:
Perhaps instead of "co-opting" `_handle_logging` in this way, which is a little different than how it was intended, we can create a separate function for this? I know it will have a little code duplication, but we can also reuse that separate function in `_handle_logging` if we want.